app/controllers/associations.js

```diff
@@ -299,3 +297,29 @@ export default class IndexController extends Controller {
     URL.revokeObjectURL(href);
   };
 }
+
+// This fetches the postal code records based on the postalCodes query param
+function PostalCodes(store, postalCodesQP) {
```

Comment:
First time using "resources" so maybe I'm doing something wrong here. It seems that the resource is recreated every time a query param changes and I'm not sure how to fix it.

This results in a "glitch" where the field is briefly emptied and then populated with the selected value again, every time a query param changes.

Not ideal, but it does fix the clear issue so better than nothing. I  need to spend more time to find a better solution.